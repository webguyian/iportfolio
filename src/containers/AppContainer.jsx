import React from 'react';
import { Route, Switch, useHistory } from 'react-router-dom';

import Button from 'components/Button/Button';
import DateTime from 'components/DateTime/DateTime';
import DeviceFrame from 'components/DeviceFrame/DeviceFrame';

import routes from 'routes';

const LeftIndicator = () => {
  const history = useHistory();
  const { pathname } = history.location;
  const home = 'home';
  const homeRoute = `/${home}`;
  const icon = pathname === homeRoute ? 'lock-open' : home;

  const redirect = () => {
    const route = icon === home ? homeRoute : '/';

    history.push(route);
  };

  if (pathname !== '/') {
    return (
      <span className="device-time-label">
        <DateTime format="h:mm" />
        <Button icon={icon} onClick={redirect}>
          Unlocked
        </Button>
      </span>
    );
  }

  return 'IMAC';
};

const AppContainer = () => {
  return (
    <div className="iportfolio-app">
      <DeviceFrame
        className="iportfolio-app-device"
        leftIndicator={<LeftIndicator />}
      >
        <Switch>
          {routes.map(route => (
            <Route key={route.pathKey} exact {...route} />
          ))}
        </Switch>
      </DeviceFrame>
    </div>
  );
};

export default AppContainer;
