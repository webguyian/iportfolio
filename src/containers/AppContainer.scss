.iportfolio-app {
  text-align: center;

  @include responsive($width-medium) {
    height: auto;
    margin-top: 5vh;
  }

  &-view {
    height: 100%;
    padding-top: $gutter-m;
    background-color: $gray-dark;
  }
}
