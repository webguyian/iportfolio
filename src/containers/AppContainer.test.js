import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import AppContainer from './AppContainer';

describe('<AppContainer />', () => {
  Date.now = jest.fn(() => new Date('2019-10-01T11:11:00'));
  window.matchMedia = jest.fn().mockImplementation(media => {
    return {
      matches: false,
      media
    };
  });

  it('renders correctly', () => {
    const component = renderer.create(
      <MemoryRouter>
        <AppContainer />
      </MemoryRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
