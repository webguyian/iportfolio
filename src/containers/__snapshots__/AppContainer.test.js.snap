// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppContainer /> renders correctly 1`] = `
<div
  className="iportfolio-app"
>
  <div
    className="iportfolio-app-device device device-iphone-x device-black"
  >
    <div
      className="device-frame"
    >
      <div
        className="device-content"
      >
        <div
          className="device-header-container"
        >
          <div
            className="device-header-bars"
          >
            <div
              className="device-header-bars-block device-header-bars-block--time"
            >
              IMAC
            </div>
            <div
              className="device-header-bars-block device-header-bars-block--icons"
            >
              <i
                className="fas icon icon-signal fa-signal"
              />
              <i
                className="fas icon icon-wifi fa-wifi"
              />
              <i
                className="fas icon icon-battery-full fa-battery-full"
              />
            </div>
          </div>
        </div>
        <div
          className="iportfolio-lockscreen"
          onTransitionEnd={[Function]}
        >
          <header
            className="iportfolio-lockscreen-header"
          >
            <i
              className="fas icon icon-lock fa-lock fa-2x"
            />
            <h1
              className="ui-clock"
            >
              11:11
            </h1>
            <h1
              className="ui-date"
            >
              Tuesday, October 1
            </h1>
          </header>
          <label
            className="ui-toggle-switch"
          >
            <input
              checked={false}
              className="ui-toggle-switch-checkbox"
              onChange={[Function]}
              type="checkbox"
            />
            <button
              className="ui-btn ui-toggle-switch-slider"
              onTransitionEnd={[Function]}
              type="button"
            />
            <span
              className="ui-toggle-switch-label"
            >
              slide to unlock
            </span>
          </label>
        </div>
      </div>
    </div>
    <div
      className="device-stripe"
    />
    <div
      className="device-header"
    />
    <div
      className="device-sensors"
    />
    <div
      className="device-btns"
    />
    <div
      className="device-power"
    />
  </div>
</div>
`;
